flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -1286,10 +1289,38 @@ private void transferAllDataFromStateHandles(
 			Map<StateHandleID, StreamStateHandle> stateHandleMap,
 			Path restoreInstancePath) throws IOException {
 
+			ExecutorService executorService = Executors.newFixedThreadPool(5);
```

Comment:
Cancelation of `copyStateDataHandleData` futures and `executorService.shutdownNow()` should be registered in `cancelStreamRegistry.registerCloseable()` and unregistered with `unregisterCloseable` before `executorService.shutdown()` in `finally`. Similar to streams in `copyStateDataHandleData`. This way we guarantee freeing of all resources in case of job abrupt shutdown.